src/components/tasks/card/index.tsx

```diff
@@ -456,6 +474,10 @@ const Card: FC<CardProps> = ({
                             progress={progress}
                             handleSaveProgressUpdate={handleSaveProgressUpdate}
                             handleProgressUpdate={handleProgressUpdate}
+                            loading={
+                                isLoadingUpdateTaskDetails ||
+                                isLoadingSelfTaskUpdate
+                            }
                         />
                     )}
                 </div>
```

Comment:
As a future refactor, please move the progress bar code separately so that all local state and variables are only scoped to that specific component.